egs/wsj/s5/utils/make_lexicon_fst.pl

```diff
@@ -148,18 +135,16 @@ sub is_sil {
         $word_or_eps = "<eps>";
         $pron_cost_string = ""; $pron_cost = 0.0; # so we only print it the 1st time.
         $s = $ns;
+      } elsif ($p ne $silphone) {
```

Comment:
Funny it did work both ways w/o warnings, despite the `use warnings` at the beginning. 
```
kkm@yupana:~/work/kaldi/egs/librispeech/s5[is_sil-fix]$ utils/make_lexicon_fst.pl data/local/dict/lexicon.txt >afternosil.txt
kkm@yupana:~/work/kaldi/egs/librispeech/s5[is_sil-fix]$ utils/make_lexicon_fst.pl data/local/dict/lexicon.txt 0.42 SIL >after.txt
kkm@yupana:~/work/kaldi/egs/librispeech/s5[is_sil-fix]$ perl --version

This is perl 5, version 18, subversion 2 (v5.18.2) built for x86_64-linux-gnu-thread-multi
(with 41 registered patches, see perl -V for more detail)
```
Added the `defined` test just in case, anyway. Thanks for spotting it.